zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -865,6 +865,29 @@ angular.module('zeppelinWebApp').controller('ParagraphCtrl', function($scope, $r
     }
   };
 
+  $scope.parseTableCell = function(cell) {
+    //here we try to automatically parse the data type for each
+    //cell in the table, i.e. string/number/date
+    // 1. is this a number (possibly)?
+    if (!isNaN(cell)) {
+      return cell.length === 0 ? cell : ((Number(value) === Number.NaN || Number(value) > Number.MAX_SAFE_INTEGER || Number(value) < Number.MIN_SAFE_INTEGER) ? cell : Number(cell));
+    }
+    // 2. is this a date?
```

Comment:
perhaps we need more flexibility on date format?
Someone's locale settings could put a format like 'DD-MM-YYYY HH:mm:ss' and it wouldn't fit any in this list, and more so, it's hard to list out all permutations here?